### ISSUE TITLE
Bad hammerhead matriarch behavior

### ISSUE THREAD

**Issue by Udrakan** [Regalis11/Barotrauma#5449]
**Description**
Hammerhead matriarch keeps escaping away from the sub inside a cave system and then when approached by divers it keeps escaping further and further away, making her extremely annoying enemy to fight.

Here is the save where it happened on multiplayer:
[BigIron2.zip](https://github.com/Regalis11/Barotrauma/files/6272985/BigIron2.zip)

**Steps To Reproduce**
-Load save, approach the hammerhead swarm from bottom
-the hammerheads attack properly, as usual
-matriarch keeps escaping and escaping away and away, it doesnt approach agro items or divers, it keeps moving away

**Version**
Unstable

**Additional information**
The hammerhead spawn do not follow active handheld beacon or flare and go for the diver
The spinelings do not follow active handheld beacon or flare and go for the diver
Might be intentional, we found out the hard way when we went diving to kill the matriarch, feels not good.

**Comment by itchyOwl** [comment]
Related Regalis11/Barotrauma-development#1373.

**Comment by Udrakan** [comment]
https://www.reddit.com/r/Barotrauma/comments/mok3ee/how_we_are_supposed_to_complete_this_mission/
Same thing hapenned to this guy from what I've read

**Comment by itchyOwl** [comment]
I changed the fleeing so that matriarch now only escapes briefly when being shot at. So it should now be possible the catch up with it and shoot it in the nose. The fix is included in the next unstable. (Regalis11/Barotrauma-development@00ded63).

It's intentional that the the matriarch attempts to avoid the player. Both the sub and the divers. The behavior could be improved upon, but we can't just make it attack straight towards the player/sub, because that would just make it a self-destructing homing missile (which it actually first was).

Handheld beacon is not labeled as "provocative", so it's ignored by monsters not explicitly configured to target "sonar". Changed this, but it's not going to be included in the next unstable or the next update. (Regalis11/Barotrauma-development@60e125c).

Couldn't repro hammerhead spawn or spineling not targeting flares.

![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/0f1728ec-7615-401e-85c9-80964be5b4fd)
